magellan-sample-advanced/src/main/java/com/wealthfront/magellan/sample/advanced/tide/TideDetailsScreen.java

```diff
@@ -32,10 +34,12 @@ public int compare(Observation o1, Observation o2) {
   };
 
   @Inject NoaaApi noaaApi;
+  @Inject @Named("LegacyNavigator") Navigator navigator;
```

Comment:
Navigating from one screen to another will now require this injected navigator. You can navigate to both old screens or the new navigables.

Reply:
Let's think through this real quick.

If you're going from `Screen` -> `Journey`, you have to provide the `Journey` with its navigational dependencies, i.e. the navigation lambdas. Old magellan `Screen`s aren't set up to do this properly, since they aren't encapsulated properly and would instead end up passing a lambda based on the global navigator, but controlled by a `Screen` which is no longer showing. The more proper way of modeling this in new magellan is to have the container in which the `Step` resides providing the navigation lambdas; at the top level, that means the `Expedition` is responsible for providing them. This could be accomplished by a top-level `Router`, or something more specific than a general `Navigator`; something that has a `goToSpecificJourney()` method that provides it with the navigation dependencies it needs.

None of this so far is incompatible with what we have already here.

Now let's look at `Journey` -> `Screen`. In this model, the `Screen` is added to the `Journey`'s backstack like any `Step` would. The problem is that `Screen`s have a very different navigational model than `Step`s; `Screens` do all navigation at the top level, which breaks the expectation that `Journey`s will continue to be `show`n until they themselves navigate away. It also breaks the mental model of `Journey`s by effectively adding a goto statement in the middle of our flow that may or may not ever return.

This mismatch of navigation expectations is what I'm most worried about, and I'm worried that it's impossible to avoid if we make `Screens` implement `Navigable`.

(Oh, also, for the same reason that I think `Screen`s being able to interrupt a `Journey` is dangerous, I think exposing a "LegacyNavigator" is a dangerous pattern if it spreads into our `Journey`s/`Step`s.)

Reply:
Agreed. Removed the injected navigator and exposing a legacy navigator through the screens which can only navigate at the expedition level. Our linear navigator forbids navigating to our old screens.